### PR TITLE
Feature | Related Content Additional Options

### PR DESCRIPTION
**Pull request by pauline-n**
#1147

### CODE REVIEW

**Review by edwinmp — CHANGES_REQUESTED** [review]
Add this to every publication page type that has related links ... there's several publication page types besides `PublicationPage`

File: di_website/publications/models.py [hunk 1]
```diff
@@ -486,16 +485,26 @@ def save(self, *args, **kwargs):
...
+            queryset = PublicationPage.objects.filter(topics__in=self.topics)
+            context['related_pages'] = get_related_pages(
+                self, self.publication_related_links.all(), queryset)
+        elif related_options == 'Country':
+            queryset = PublicationPage.objects.filter(page_countries__in=self.page_countries)
+            context['related_pages'] = get_related_pages(
+                self, self.publication_related_links.all(), queryset)
+        else:
+            context['related_pages'] = get_related_pages(
+                self, self.publication_related_links.all(), PublicationPage.objects)
```
Comment (edwinmp) [inline di_website/publications/models.py:500]: You have this almost exactly repeated in like 5 different places ... Use a function and add it in the `RelatedLinksMixin`

```
def get_related_pages(self, option='Manual', PageClass):
     ....
```

Usage

`self.get_related_pages(related_options, PublicationPage)` - be sure to replace PublicationPage with all the other page types as required.

**Review by edwinmp — CHANGES_REQUESTED** [review]

File: di_website/publications/models.py (hunk 1 above)
Comment (pauline-n) [reply di_website/publications/models.py:500]: Great, thanks

**Review by edwinmp — CHANGES_REQUESTED** [review]
The country option doesn't seem to work - the links shown aren't for the target country.

Also, got an error when trying to view some of the publications.

image

**Review by edwinmp — CHANGES_REQUESTED** [review]
The country option doesn't seem to work - the links shown aren't for the target country.

Also, got an error when trying to view some of the publications.

image

**Review by edwinmp — APPROVED** [review]
No idea if this works, so may as well get it to staging and find out :)

**Pull request merged**